### PR TITLE
Fixed AutoAttack stop working with AutoEat enabled and no food on hotbar

### PR DESCRIPTION
**Pull request by ReinforceZwei**
#934

### CODE REVIEW

File: MinecraftClient/ChatBots/AutoEat.cs [hunk 1]
```diff
@@ -20,15 +21,22 @@ public AutoEat(int Threshold)
...
-                FindFoodAndEat();
+                Eating = FindFoodAndEat();
+                if (!Eating)
+                    ChangeSlot(LastSlot);
             }
             // keep eating until full
             if (food < 20 && Eating)
             {
-                FindFoodAndEat();
+                Task.Factory.StartNew(delegate
```
Comment (ORelio) [inline MinecraftClient/ChatBots/AutoEat.cs:33]: Unfortunately, MCC is not fully thread safe and might cause an exception when using the ChatBot API from a different thread. As a workaround, you can set a boolean like `EatOnNextUpdate = true` (or use a counter decremented each tick), then do the logic in `OnUpdate()`. Apologies for the low code quality of MCC's core 😅

Comment (ReinforceZwei) [reply MinecraftClient/ChatBots/AutoEat.cs:33]: Don’t mind. I have came up with a better solution that do not require Threading. I will modify and test the code tomorrow.

**Comment by ReinforceZwei** [comment]
My idea is decrease player's inventory food count by one after found a food so that we don't need to wait for the server to *update* our inventory, but `ChatBot.GetPlayerInventory()` return a **copy** of player inventory, hence I cannot modify the player inventory inside a ChatBot. Should I add a new API or make that method return the reference of player inventory?

**Comment by ORelio** [comment]
I think it could pose other issues if the player inventory no longer reflect what's really in the player inventory server-side. Another bot could try to use an "empty" slot that actually contains food. However, the AutoEat bot could _store_ its copy of the inventory, or store slot ID and item count, then modify its own copy.

**Comment by ReinforceZwei** [comment]
It might become quite complicated if AutoEat store its own copy of inventory as it require some work on synchronize the inventory with both McTcpClient and server. I think I will just use the `OnUpdate()` as the solution.

**Pull request merged**